analyzer/src/funTest/assets/projects/synthetic/yarn-workspaces-expected-output.yml

```diff
@@ -19,7 +19,13 @@ project:
   homepage_url: ""
   scopes:
   - name: "dependencies"
-    dependencies: []
+    dependencies:
+    - id: "NPM:@here:harp-fetch:0.11.0"
+      dependencies:
+      - id: "NPM::node-fetch:2.6.0"
+    - id: "NPM:@here:harp-fetch:0.3.6"
+      dependencies:
+      - id: "NPM::node-fetch:2.6.0"
```

Comment:
Should the workspace packages like "NPM::pkg1:1.0.0" not also appear in the dependency tree at the place of the symbolic link? Also, shouldn't the package be called "Yarn::pkg1:1.0.0" instead?

Reply:
1. It might  be better if `pkg1` it appeared in the dependency tree. I would like to draw a line here and keep this as potential future improvement if you are ok with that. E.g. I don't see a big issue with this and also the workspace submodule structure is collapsed for now anyway. Are you ok with that?
2. Naming the package "Yarn::pkg1:1.0.0" instead of "NPM..." I  would also leave it up to a future improvement.
Note that it should be out of scope of this change as it does not touch the `Package` instance creation at all.

Reply:
1. I'm fine with keeping it as is in this PR but this is something that should be fixed as soon as possible. Because it makes it unnecessarily hard for the user to understand how the dependencies were resolved, and it could potentially lead to wrong evaluation results if `DependencyRule`s are used in the evaluator.

Reply:
2. Ok

Reply:
Alright, I'll do a follow up PR for 1 then later on.

Reply:
Thanks.